lib/lib_rss.php

```diff
@@ -397,7 +406,20 @@ function enforceHttpEncoding(string $html, string $contentType = ''): string {
 		// Existing XML declaration, do nothing
 		return $html;
 	}
-	return '<' . '?xml version="1.0" encoding="' . $httpCharsetNormalized . '" ?' . ">\n" . $html;
+	if ($httpCharsetNormalized !== 'UTF-8') {
+		// Try to change encoding to UTF-8 using mbstring or iconv or intl
+		$utf8 = SimplePie_Misc::change_encoding($html, $httpCharsetNormalized, 'UTF-8');
+		if (is_string($utf8)) {
+			$html = stripHtmlMetaCharset($utf8);
+			$httpCharsetNormalized = 'UTF-8';
+		}
+	}
+	if ($httpCharsetNormalized === 'UTF-8') {
+		// Save encoding information as XML declaration
+		return '<' . '?xml version="1.0" encoding="' . $httpCharsetNormalized . '" ?' . ">\n" . $html;
+	}
+	// Give up
```

Comment:
Glancing at this I was wondering if it might make more sense to give up with the charset from the `<meta>` element?

Reply:
`DOMDocument::loadHTML()` will already sniff information from `<meta charset`. This is the reason we have to strip this information after conversion to UTF-8.
Furthermore, we cannot always write `<?xml encoding`, in particular for character encodings that are not supersets of ASCII, such as UTF-32.
So as far as this function goes, I do not believe we can do much more.